polaris/hub/client.py

```diff
@@ -610,28 +610,24 @@ def upload_dataset(
         # Step 3: Upload any associated Zarr archive
         if dataset.zarr_root is not None:
             with tmp_attribute_change(self.settings, "default_timeout", timeout):
-                # Copy the Zarr archive to the hub
-                dest = self.open_zarr_file(
-                    owner=dataset.owner,
-                    name=dataset.name,
-                    path=dataset_json["zarrRootPath"],
-                    mode="w",
-                    as_consolidated=False,
+                polaris_fs = PolarisFileSystem(
+                    polaris_client=self,
+                    dataset_owner=dataset.owner,
+                    dataset_name=dataset.name,
                 )
+                dest = zarr.storage.FSStore(url=dataset_json["zarrRootPath"], fs=polaris_fs, mode="w")
 
                 # Locally consolidate Zarr archive metadata. Future updates on handling consolidated
                 # metadata based on Zarr developers' recommendations can be tracked at:
                 # https://github.com/zarr-developers/zarr-python/issues/1731
                 zarr.consolidate_metadata(dataset.zarr_root.store.store)
```

Comment:
Since we by now assume a Zarr archive is always consolidated, we can remove these lines, right?